src/components/SearchBar.js

```diff
@@ -219,8 +219,12 @@ class SearchBar extends React.Component<SearchBarDefaultProps, SearchBarProps, S
     }
     if (searcher) {
       if (doSearch) {
-        searcher.setQueryAndSearch(newQuery);
-        this.route();
+        if (!searcher.state.haveSearched) {
```

Comment:
Please add a comment here explaining what happens and in which cases...

Reply:
Done!